openupgrade_scripts/scripts/mail/15.0.1.5/post-migration.py

```diff
@@ -0,0 +1,55 @@
+from openupgradelib import openupgrade
+
+
+def _map_activity_type_chaining_type_field(env):
+    openupgrade.map_values(
+        env.cr,
+        openupgrade.get_legacy_name("force_next"),
+        "chaining_type",
+        [("true", "trigger"), ("false", "suggest")],
+        table="mail_activity_type",
+    )
+
+
+def _map_activity_type_res_model_field(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mail_activity_type
+        SET res_model = ir_model.model
+        FROM ir_model
+        WHERE mail_activity_type.%s = ir_model.id
+        """
+        % openupgrade.get_legacy_name("res_model_id"),
+    )
+
+
+def _map_mail_notification_failure_type(env):
+    openupgrade.map_values(
+        env.cr,
+        openupgrade.get_legacy_name("failure_type"),
+        "failure_type",
+        [
+            ("SMTP", "mail_smtp"),
+            ("RECIPIENT", "mail_email_invalid"),
+            ("UNKNOWN", "unknown"),
```

Comment:
I think you should also handle BOUNCE, otherwise there can be invalid entries left after the migration. I would suggest `mail_email_invalid` or `unknown`.

Reply:
thanks, i will add it